src/middlewared/middlewared/plugins/audit/audit.py

```diff
@@ -190,6 +199,103 @@ async def query(self, data):
 
         return filter_list(results, data['query-filters'], data['query-options'])
 
+    @accepts(Patch(
+        'audit_query', 'audit_export',
+        ('add', Str('export_format', enum=['CSV', 'JSON', 'YAML'], default='JSON')),
+    ))
+    @returns(Str('audit_file_path'))
+    @pass_app()
+    def export(self, app, data):
```

Comment:
So we have a limit option that can be passed? Do we expect to honor that here?

Lastly, are we concerned about the amount of data that can be queried and then written to disk? Would this potentially block for a large amount of time based on the number of entries that are being exported? Just food for thought